### PR TITLE
Replace deep-equal with same-object lib

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "error-stack-parser": "^2.1.4",
     "glob": "^8.0.3",
     "minimist": "^1.2.6",
+    "same-object": "^1.0.0",
```
Comment (voxpelli) [inline package.json:26]: You probably want to remove `deep-equal` as well?

**Comment by voxpelli** [comment]
Also: The URL:s to GitHub doesn't work here: https://www.npmjs.com/package/same-object

And I'm not sure which benefits this gives over the library it replaces? Is it a personal preference or something else?

File: package.json (hunk 1 above)
Comment (LuKks) [reply package.json:25]: Thanks! Actually did but didn't push!

**Comment by LuKks** [comment]
@voxpelli Now the repository is public

**Comment by LuKks** [comment]
`deep-equal` doesn't run on the internal JavaScript runtime which it doesn't have any native Node modules like `util`, etc.

`same-object` aims to provide "99%" compatibliity with `deep-equal` without requiring native dependencies. It's based on `deep-eql`, but will probably end up forking to quickly extend support if needed.

**Comment by LuKks** [comment]
Tests from `deep-equal` were copied into `same-object` and most of them passes (except very edge cases atm, which is ok).

Also tested this `brittle` branch with lots of repos like hypercore, hyperbee, framed-stream, libudx, hyperswarm, dht, and many others, so all seems good!

**Comment by voxpelli** [comment]
@LuKks Ah, I see! Would be great to note that reason in the same-object readme 👍

**Review by voxpelli — APPROVED** [review]

**Comment by LuKks** [comment]
Notes added to `same-object` README!
